src/format.iced

```diff
@@ -0,0 +1,115 @@
+util = require('./util')
+
+# Punctuation - this is modular
+space = new Buffer(' ')
+newline = new Buffer('\n')
+punctuation = new Buffer('.')
+
+words_per_line = 200
+chars_per_word = 15
+
+# For working with the chunk streams
+noop = () ->
+
+# This stream takes basex'd input and writes framed output, as per https://saltpack.org/armoring#framing-the-basex-payload
+exports.FormatStream = class FormatStream extends stream.ChunkStream
+
+  # _format is the transform function passed to the chunk stream constructor
+  _format : (chunk, cb) ->
+    # this is only ever called during flush()
+    if chunk.length < @block_size
+      return cb(null, chunk)
+
+    # write out the header if we haven't already
+    res = new Buffer('')
+    unless @_header_written
+      res = Buffer.concat([@_header, punctuation, space])
+      @_header_written = true
+
+    @_word_count++
+    if @_word_count isnt 200
+      # res will be an empty buffer except when writing the header
+      res = Buffer.concat([res, chunk, space])
+    else
+      # ditto
+      res = Buffer.concat([res, chunk, newline])
+      @_word_count = 0
+
+    return cb(null, res)
+
+  flush_append : (cb) ->
+    return cb(null, Buffer.concat([punctuation, space, @_footer, punctuation]))
+
+  constructor : ({brand}) ->
+    if brand? then _brand = brand else _brand = 'KEYBASE'
+    @_header = new Buffer("BEGIN#{space}#{_brand}#{space}SALTPACK#{space}ENCRYPTED#{space}MESSAGE")
+    @_footer = new Buffer("END#{space}#{_brand}#{space}SALTPACK#{space}ENCRYPTED#{space}MESSAGE")
+    @_header_written = false
+    @_word_count = 0
+    super({transform_func : @_format, block_size : chars_per_word, exact_chunking : true, writableObjectMode : false, readableObjectMode : false})
+
+exports.DeformatStream = class DeformatStream extends stream.ChunkStream
+
+  _header_mode = 0
+  _body_mode = 1
+  _footer_mode = 2
+  _strip_re = /[>\n\r\t ]/g
+
+  _strip = (chunk) -> chunk = new Buffer(chunk.toString().replace(_strip_re, ""))
+
+  _deformat : (chunk, cb) ->
+    chunk = Buffer.concat([@_partial, chunk])
+    @_partial = new Buffer('')
+    if @_mode is _header_mode
+      index = chunk.indexOf(punctuation[0])
+      if index isnt -1
+        # then we have a full header, verify it
+        re = /[>\n\r\t ]*BEGIN[>\n\r\t ]+([a-zA-Z0-9]+)?[>\n\r\t ]+SALTPACK[>\n\r\t ]+(ENCRYPTED[>\n\r\t ]+MESSAGE)|(SIGNED[>\n\r\t ]+MESSAGE)|(DETACHED[>\n\r\t ]+SIGNATURE)[>\n\r\t ]*/
+        @_header = chunk[...index]
+        unless re.test(@_header) then return cb(new Error("Header failed to verify!"), null)
+        chunk = chunk[index + punctuation.length...]
+        @_mode = _body_mode
+      else
+        # then we have a partial header, store it and wait for the next write
+        @_partial = chunk
+        return cb(null, null)
+
+    if @_mode is _body_mode
+      index = chunk.indexOf(punctuation[0])
+      if index is -1
+        # chunk is only body bytes
+        return cb(null, _strip(chunk))
```

Comment:
Well it does, but in the deformatter, we don't care about length. Now that you mention it, it would probably make more sense to have `DeformatStream` just extend a standard transform stream, not `ChunkStream`.